DataverseGen.Core/Metadata/MappingField.cs

```diff
@@ -227,20 +110,40 @@ public static MappingField Parse(AttributeMetadata attribute, MappingEntity enti
             if (attribute.DisplayName?.UserLocalizedLabel != null)
                 result.Label = attribute.DisplayName.UserLocalizedLabel.Label;
 
-            result.IsRequired = attribute.RequiredLevel != null && attribute.RequiredLevel.Value == AttributeRequiredLevel.ApplicationRequired;
+            result.IsRequired = attribute.RequiredLevel != null &&
+                                attribute.RequiredLevel.Value == AttributeRequiredLevel.ApplicationRequired;
 
             result.Attribute =
                 new CrmPropertyAttribute
                 {
                     LogicalName = attribute.LogicalName,
-                    IsLookup = attribute.AttributeType == AttributeTypeCode.Lookup || attribute.AttributeType == AttributeTypeCode.Customer
+                    IsLookup = attribute.AttributeType == AttributeTypeCode.Lookup ||
+                               attribute.AttributeType == AttributeTypeCode.Customer
                 };
             result.TargetTypeForCrmSvcUtil = GetTargetType(result);
             result.FieldTypeString = result.TargetTypeForCrmSvcUtil;
 
             return result;
         }
 
+        public MappingField CreateFileNameField(MappingField field)
+        {
+            MappingField fieldCopy = DeepCloneExtensions.CreateDeepCopy(field);
+            fieldCopy.TargetTypeForCrmSvcUtil = "string";
+            fieldCopy.DisplayName = $"{fieldCopy.DisplayName}Name";
+            fieldCopy.Attribute.LogicalName = $"{fieldCopy.Attribute.LogicalName}_name";
+            return fieldCopy;
+        }
+
+        public MappingField CreateLookupNameField(MappingField field)
+        {
+            MappingField fieldCopy = DeepCloneExtensions.CreateDeepCopy(field);
+            fieldCopy.TargetTypeForCrmSvcUtil = "string";
+            fieldCopy.DisplayName = $"{fieldCopy.DisplayName}Name";
+            fieldCopy.Attribute.LogicalName = $"{fieldCopy.Attribute.LogicalName}name";
```

Comment:
this method is the same as CreateFileNameField except line 143 ("...name" instead of "..._name")